Makefile

```diff
@@ -33,6 +33,53 @@ $(PROG): $(OBJ:%=$(BUILDDIR)/%)
 $(BUILDDIR)/%.o: %.c | builddir
 	$(CC) -MMD -c $(CFLAGS) -o $@ $<
 
+fip:
+	if [ -d "$(BUILDDIR)/fip" ]; then rm -r "$(BUILDDIR)/fip"; fi
+ifdef FIP
+	cp -r "$(FIP)/" "$(BUILDDIR)/fip"
+else
+	curl -L https://github.com/BayLibre/u-boot/releases/download/v2017.11-libretech-cc/libretech-cc_fip_20180418.tar.gz | tar -xvz -C $(BUILDDIR)
+endif
+
+$(BUILDDIR)/fip/gxl/bl2_acs.bin: fip
+	python acs_tool.py $(BUILDDIR)/fip/gxl/bl2.bin \
+		$(BUILDDIR)/fip/gxl/bl2_acs.bin \
+		$(BUILDDIR)/fip/gxl/acs.bin 0
+
+$(BUILDDIR)/fip/gxl/bl2_new.bin: fip $(BUILDDIR)/fip/gxl/bl2_acs.bin
+	$(BUILDDIR)/fip/blx_fix.sh $(BUILDDIR)/fip/gxl/bl2_acs.bin zero_tmp \
+		$(BUILDDIR)/fip/gxl/bl2_zero.bin $(BUILDDIR)/fip/gxl/bl21.bin \
+		$(BUILDDIR)/fip/gxl/bl21_zero.bin $(BUILDDIR)/fip/gxl/bl2_new.bin bl2
+
+$(BUILDDIR)/fip/gxl/bl2.bin.enc: $(BUILDDIR)/fip/gxl/bl2_new.bin
+	./$(PROG) -t bl2 -c $(BUILDDIR)/fip/gxl/bl2_new.bin $(BUILDDIR)/fip/gxl/bl2.bin.enc
+
+$(BUILDDIR)/fip/gxl/bl30_new.bin: $(BUILDDIR)/fip/gxl/bl30.bin
+	$(BUILDDIR)/fip/blx_fix.sh $(BUILDDIR)/fip/gxl/bl30.bin zero_tmp \
+	$(BUILDDIR)/fip/gxl/bl30_zero.bin $(BUILDDIR)/fip/gxl/bl301.bin \
+	$(BUILDDIR)/fip/gxl/bl301_zero.bin $(BUILDDIR)/fip/gxl/bl30_new.bin bl30
+
+$(BUILDDIR)/fip/gxl/bl30.bin.enc: $(BUILDDIR)/fip/gxl/bl30_new.bin
+	./$(PROG) -t bl3 -c $(BUILDDIR)/fip/gxl/bl30_new.bin $(BUILDDIR)/fip/gxl/bl30.bin.enc
+
+$(BUILDDIR)/fip/gxl/bl31.bin.enc: fip
+	./$(PROG) -t bl3 -c $(BUILDDIR)/fip/gxl/bl31.img $(BUILDDIR)/fip/gxl/bl31.img.enc
+
+$(BUILDDIR)/fip/gxl/u-boot.bin.enc:
```

Comment:
Pleas make this target depending of $(UBOOT) and gxlimg prerequisite so u-boot.bin.enc won't be re-do if $(UBOOT) and gxlimg has not changed